<?xml version="1.0"?>
<robot name="sailboat">
	<link name="world"/>
	<link name="base_hull">
		<visual>
			<origin rpy="1.57 0 0"/>
			<geometry>
				<cylinder length="2.0" radius="0.1"/>
			</geometry>
		</visual>
	</link>
	<link name="base_rudder"/>
	<link name="base_sail"/>
	<link name="sailboat_sail">
                <visual>
			<origin xyz="0 -0.25 0"/>
                        <geometry>
                                <box size="0.01 0.5 1"/>
                        </geometry>
                </visual>
        </link>

	<link name="sailboat_rudder">
                <visual>
			<origin xyz="0 -0.05 0"/>
                        <geometry>
                                <box size="0.01 0.1 0.2"/>
                        </geometry>
                </visual>
        </link>
	<joint name="world_hull" type="floating">
		<parent link="world"/>
		<child link="base_hull"/>
	</joint>

	<joint name="hull_rudder" type="fixed">
		<parent link="base_hull"/>
		<child link="base_rudder"/>

		<origin xyz="0 -0.4 0"/>
	</joint>

        <joint name="hull_sail" type="fixed">
                <parent link="base_hull"/>
                <child link="base_sail"/>

		<origin xyz="0 0.05 0" />
        </joint>

        <joint name="rudder_angle" type="revolute">
                <parent link="base_rudder"/>
                <child link="sailboat_rudder"/>

		<origin xyz="0 0 -0.1"/>
		<axis xyz="0 0 1"/>
                <limit lower="-0.78" upper="0.78" effort="100" velocity="1"/>
        </joint>

        <joint name="sail_angle" type="revolute">
                <parent link="base_sail"/>
                <child link="sailboat_sail"/>

		<origin xyz="0 0 0.5"/>
		<axis xyz="0 0 1"/>
		<limit lower="-1.57" upper="1.57" effort="100" velocity="1"/>
        </joint>

</robot>
